(TK-430) Migrate to clj-parent 0.3.3 and bump i18n to 0.6.0

camlow325 opened the pull request:
This PR migrates tk-filesystem-watcher to using clj-parent version 0.3.3
for its dependencies.  This commit also adds i18n as a plugin dependency, to
allow for message POT files to be regenerated as needed in CI.  This
commit also upgrades the lein version to 2.7.1 in the .travis.yml
file since 2.7.1 is the minimum version required for compatibility
with lein/clj-parent.

This PR also removes tk-scheduler as a dependency since none of the
production code was referencing it and the tests that were referencing it
did not seem to be needing to do so.

camlow325 committed:
(TK-430) Migrate to clj-parent 0.3.3 and bump i18n to 0.6.0

This commit migrates tk-filesystem-watcher to using clj-parent version 0.3.3
for its dependencies.  This commit also adds i18n as a plugin dependency, to
allow for message POT files to be regenerated as needed in CI.  This
commit also upgrades the lein version to 2.7.1 in the .travis.yml
file since 2.7.1 is the minimum version required for compatibility
with lein/clj-parent.

camlow325 committed:
(TK-430) Removed tk-scheduler dependency

This commit removes tk-scheduler as a project dependency.  Previously,
it had only been used in a test namespace, where it did not appear to be
needed in order for the tests to pass.

jpinsonault commented on project.clj:
```diff
@@ -28,12 +29,15 @@
                                    :classifier "test"
                                    :scope "test"]
-                                  [puppetlabs/kitchensink ~ks-version
+                                  [puppetlabs/kitchensink nil
```
Does putting `nil` here mean it'll pull the version from the parent project? Sucks that we have to leave a `nil` lying around

camlow325 replied:
Yeah, the `nil` would signify that that the version is pulled in from the parent project.  It may actually work without the `nil` in later lein-parent versions, can't remember.  I can try removing it and see if it still works.

camlow325 replied:
Yeah, turns out the `nil` references don't appear to be needed after all.  I removed those in 45414655.

camlow325 committed:
(TK-430) Remove nil from tk and kitchensink test util dependencies

This commit removes the superfluous `nil` references from the tk and
kitchensink test util dependencies.

camlow325 committed:
(MAINT) Remove openjdk7 from trapperkeeper-filesystem-watcher travis.yml

This commit removes openjdk7 from the trapperkeeper-filesystem-watcher
travis.yml testing matrix.

puppetcla commented:
CLA signed by all contributors.
